src/main/java/iterator/RichIterator.java

```diff
@@ -477,7 +905,33 @@ default RichIterator<A> distinct() {
      * @return the iterator
      */
     default <B> RichIterator<B> flatMap(Function<? super A, ? extends Iterator<B>> f) {
-        throw new NotImplementedException();
+        return new RichIterator<B>() {
+            private Iterator<B> iterator;
+
+            @Override
+            public boolean hasNext() {
+                if (iterator != null && iterator.hasNext()) {
```

Comment:
The null check can be avoided by initiating the `iterator` data member to `empty()`